src/MonC/Artifact.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.IO;
+
+namespace MonC
+{
+    /// <summary>
+    /// Abstract compiler artifact which may potentially own unmanaged resources
+    /// </summary>
+    public abstract class Artifact : IDisposable
```

Comment:
This might be better suited as an interface, as there's no base implementation here.